### ISSUE TITLE
Astropy/vo/samp uses ssl.PROTOCOL_SSLv2 and ssl.PROTOCOL_SSLv3

### ISSUE THREAD

**Issue by olebole** [astropy/astropy#3306]
The samp package seems to use SSL version 2 and 3 which had [serious security flaws discovered](https://www.openssl.org/~bodo/ssl-poodle.pdf) in the last months. On Debian, both protocols are disabled now; therefore the compilation fails, specifically `astropy/tests/tests/test_imports.py:27: test_imports FAILED`.

The errors are like
```
_________________ ERROR collecting astropy/vo/samp/__init__.py _________________
astropy/vo/samp/__init__.py:15: in <module>
>   from .hub import *
astropy/vo/samp/hub.py:33: in <module>
>       from .ssl_utils import SafeTransport, SecureXMLRPCServer
astropy/vo/samp/ssl_utils.py:26: in <module>
>       class HTTPSConnection(HTTPConnection):
astropy/vo/samp/ssl_utils.py:35: in HTTPSConnection
>                    ssl_version=ssl.PROTOCOL_SSLv3, strict=None):
E           AttributeError: 'module' object has no attribute 'PROTOCOL_SSLv3'
```

**Comment by mdboom** [comment]
@olebole: I'll take a look.  Do you have a reference to the changes Debian has made to the package?

**Comment by mdboom** [comment]
@olebole: Nevermind.  I see how this works.  Python 2.7.9 hides SSLv3 if the underlying openssl doesn't have it (I was looking at an earlier version of Python which doesn't have this check).

**Comment by embray** [comment]
Yeah, I don't know why this wouldn't just use TLS.

**Comment by embray** [comment]
I'd be willing to make a patch release with this fix as well since this affects the 0.4.x branch as well.

**Comment by mdboom** [comment]
We can't just use TLS because TLS 1.1 and 1.2 are only available in Python 2.7.9 and 3.4.x.  So best to just let Python "recommend" what to use by default, and if the user cares, they'll have to install a version of Python that works for their needs anyway...

**Comment by embray** [comment]
Makes sense.

**Comment by embray** [comment]
Fixed by #3308.
